torchvision/transforms/functional_tensor.py

```diff
@@ -236,3 +276,51 @@ def _blend(img1, img2, ratio):
     # type: (Tensor, Tensor, float) -> Tensor
     bound = 1 if img1.dtype in [torch.half, torch.float32, torch.float64] else 255
     return (ratio * img1 + (1 - ratio) * img2).clamp(0, bound).to(img1.dtype)
+
+
+def _rgb2hsv(img):
+    r, g, b = img[0], img[1], img[2]
+
+    maxc, _ = torch.max(img, dim=0)
+    minc, _ = torch.min(img, dim=0)
+    uv = maxc
+
+    cr = maxc.to(dtype=torch.float32) - minc.to(dtype=torch.float32)
+    s = cr / maxc.to(dtype=torch.float32)
+    rc = (maxc-r).to(dtype=torch.float32) / cr
+    gc = (maxc-g).to(dtype=torch.float32) / cr
+    bc = (maxc-b).to(dtype=torch.float32) / cr
```

Comment:
If we move the casting to float in the beginning of the function, we can avoid casting individually each tensor.
Something like

```suggestion
    orig_dtype = img.dtype
    img = img.to(dtype=torch.float32)
    r, g, b = img.unbind(0)

    maxc, _ = torch.max(img, dim=0)
    minc, _ = torch.min(img, dim=0)
    uv = maxc

    cr = maxc - minc
    s = cr / maxc
    rc = (maxc-r) / cr
    gc = (maxc-g) / cr
    bc = (maxc-b) / cr
```

Reply:
Done.

Reply:
Done